repos/fdbt-site/src/pages/api/deleteCap.ts

```diff
@@ -15,6 +19,26 @@ export default async (req: NextApiRequestWithSession, res: NextApiResponse): Pro
 
         const noc = getAndValidateNoc(req, res);
 
+        const products = await getAllProductsByNoc(noc);
+        const matchingJsonLinks = products.map((product) => product.matchingJsonLink);
+        const tickets: TicketWithIds[] = await Promise.all(
+            matchingJsonLinks.map(async (link) => {
+                return await getProductsMatchingJson(link);
+            }),
+        );
+
+        const productsUsingCap = tickets.filter((t) => 'cap' in t && t.cap && t.cap.id === id);
+
+        if (productsUsingCap.length > 0) {
+            const cap = (await getCapByNocAndId(noc, id)) as Cap;
```

Comment:
Can we move this line right up to after we check the id is a number, and if it comes back as undefined, throw an error there. Then when we use the cap down here, we dont need to cast it to Cap.